### PR TITLE
PML-319: bugfix for the loss of the Builder GPU test

### PR DESCRIPTION
**Pull request by CassNot**
## Summary

There was a bugged GPU test for the builder. The loss of `test_entangling_layer_models_on_gpu` was logits.sum(), which is always 1 and therefore the gradients are almost null and the test was failing.
The tests started failing after tag 2.3.
The commit 3e9b3df changed default FOCK to UNBUNCHED. As UNBUNCHED is normalized, the loss is a constant and the gradient of the constant is null. When it was FOCK, because FOCK is derived from the AMPLITUDES and not normalized, even if the loss is close to 1, it was not exactly 1 (so the gradient was not completely null)

## Related Issue
   
PML-319 

## Type of change
- [x] Bug fix
- [ ] New feature
- [ ] Documentation update
- [ ] Refactor / Cleanup
- [ ] Performance improvement
- [ ] CI / Build / Tooling
- [ ] Breaking change (requires migration notes)

## Proposed changes

The loss is now the difference between the input and the logits (it does not make physical sense but the gradients are non null)

## How to test / How to run

1. Command lines

```
Block of code
```

## Documentation
- [ ] User docs updated (Sphinx)
- [ ] Examples / notebooks updated
- [ ] Docstrings updated
- [ ] Updated the API

## Checklist
- [x] PR title includes Jira issue key (e.g., PML-126)
- [x] "Related Jira ticket" section includes the Jira issue key (no URL)
- [x] Code formatted (ruff format)
- [x] Lint passes (ruff)
- [x] Static typing passes (mypy) if applicable
- [x] Unit tests added/updated (pytest)
- [x] Tests pass locally (pytest)
- [x] Tests pass on GPU (pytest)
- [x] Test coverage not decreased significantly
- [x] Docs build locally if affected (sphinx)
- [x] With this command: 

        > SPHINXOPTS="-W --keep-going -n" make -C docs clean html 

    the docs are built without any warning or errors.
- [x] New public classes/methods/packages are added in the API following the methodology presented in other files.
- [x] Dependencies updated (if needed) and pinned appropriately
- [x] PR description explains what changed and how to validate it

### CODE REVIEW

File: tests/builder/test_circuit_builder.py
Comment (LF-Vigneux) [inline tests/builder/test_circuit_builder.py:1]: Remove the copilot comment

**Review by LF-Vigneux — CHANGES_REQUESTED** [review]
Just remove the LLm comment and everything is good!

File: tests/builder/test_circuit_builder.py
Comment (CassNot) [reply tests/builder/test_circuit_builder.py:1]: I updated the comment to say that we want the loss to vary (non-trivially)

**Review by LF-Vigneux — APPROVED** [review]
The bug is indeed fixed
